docs/sdk/MetricDataRetrieval/MetricDataRetrieval.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Goal: Extract facility metric data using the Contxt Python SDK
+#
+# Contxt organizes its data in a Customer to Facility to Metric hierarchy. To retrieve metric data
+# for all facilities, the general operation flow is:
+# 1. Instance customer client
+# 1. Locate customer facilities
+# 1. Locate customer metric definitions
+# 1. Retrieve metrics for each customer facility.
+#
+# This document demonstrates this flow using the
+# [Contxt Python SDK](https://github.com/ndustrialio/contxt-sdk-python).
+#
+# Note: this example is functional but not optimized. At a minimum, multiprocessing could be used
+# to speed up the data request for a large number of facilities.
+
+# In[1]:
+
+
+from dataclasses import field, make_dataclass
+from datetime import datetime
+from random import randint
+from typing import Any
+
+import pytz
+
+from contxt.cli.clients import Clients
+
+# # How long ago should we look for data?
+#
+# The metric data retrieval service allows bounding a time window for returned data with
+# `effective_start_time` and `effective_end_time` parameters. These values must be reported as
+# ISO 8601 strings matching the `"%Y-%m-%dT%H:%M:%SZ"` format.
+#
+# In this document, we're going to retrieve all data between now and the date in the
+# `earliest_date` variable we declare below.
+
+# In[2]:
+
+
+earliest_date = datetime(2022, 1, 1, tzinfo=pytz.UTC).strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
since this is just a constant (which allows us to drop `pytz`):

```suggestion
earliest_date = "2022-01-01T00:00:00Z"
```

fun fact, `utc` is in the [stdlib](https://docs.python.org/3/library/datetime.html#datetime.timezone.utc). and starting with 3.9, the stdlib also has its own [timezone database](https://docs.python.org/3/library/zoneinfo.html)